apm-sniffer/apm-sdk-plugin/cassandra-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cassandra/v3/SessionExecuteAsyncWithStatementArgInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.cassandra.v3;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.Statement;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+/**
+ * @author stone.wlg
+ */
+public class SessionExecuteAsyncWithStatementArgInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public final void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                   Class<?>[] argumentsTypes,
+                                   MethodInterceptResult result) throws Throwable {
+        ConnectionInfo connectionInfo = (ConnectionInfo) objInst.getSkyWalkingDynamicField();
+        Statement statement = (Statement) allArguments[0];
+        String remotePeer = statement.getHost() == null ? connectionInfo.getContactPoints() : (statement.getHost().getSocketAddress().getHostName() + ":" + statement.getHost().getSocketAddress().getPort());
+        String keyspace = statement.getKeyspace() == null ? connectionInfo.getKeyspace() : statement.getKeyspace();
+        String query = statement.toString();
+        if (statement instanceof BoundStatement) {
+            query = ((BoundStatement) statement).preparedStatement().getQueryString();
+        }
+
+        AbstractSpan span = ContextManager.createExitSpan(Constants.CASSANDRA_OP_PREFIX + method.getName(), remotePeer);
+        span.setComponent(Constants.CASSANDRA_COMPONENT_NAME);
```

Comment:
Why not use this API `AbstractSpan setComponent(Component component);`?

Reply:
I have no idea about it, i need to modify ComponentDefine.java and component-libraries.yml, that  is a little bit confused to me, why do i need to change these two files ? component-libraries.yml is easy to use. Do i must to use ComponentDefine.java ?

Reply:
ok, i will use Component class instead of String for performance.